RawVideo/RawVideo/ContentView.swift

```diff
@@ -0,0 +1,1032 @@
+                            }
+                            Button(action: {
+                                Task {
+                                    await EndCall()
+                                }
+                            })
+                            {
+                                HStack {
+                                    Text("End Call")
+                                        .accentColor(.black)
+                                }
+                                .frame(width: 120, height: 30)
+                                .padding(10)
+                                .background(Color(.red))
+                                .cornerRadius(5)
+                            }
+                        }
+                    }
+                }
+                .frame(width: geometryReader.size.width, height: geometryReader.size.height)
+                .onAppear(perform: {
+                    let defaults = UserDefaults.standard
+                    let savedToken = defaults.value(forKey: "Token")
+                    
+                    if (savedToken != nil)
+                    {
+                        token = savedToken as! String
+                    }
+                })
+                .onTapGesture {
+                    if (isFocused)
+                    {
+                        isFocused = false
+                    }
+                }
+                .disabled(loading)
+                .zIndex(0)
+            }
+        }
+    }
+
+    struct ContentView_Previews: PreviewProvider 
+    {
+        static var previews: some View
+        {
+            ContentView()
+        }
+    }
+    
+    private func InitResources() async -> Void
+    {
+        localVideoStreamObserver = LocalVideoStreamObserver(view: self)
+        virtualRawOutgoingVideoStreamObserver = VirtualRawOutgoingVideoStreamObserver(view: self)
+        screenShareRawOutgoingVideoStreamObserver = ScreenShareRawOutgoingVideoStreamObserver(view: self)
+        remoteVideoStreamObserver = RemoteVideoStreamObserver(view: self)
+        rawIncomingVideoStreamObserver = RawIncomingVideoStreamObserver(view: self)
+        rawIncomingVideoStreamObserver?.delegate = OnRawVideoFrameArrived
+        remoteParticipantObserver = RemoteParticipantObserver(view: self)
+        callObserver = CallObserver(view: self, remoteParticipantObserver: remoteParticipantObserver!)
+        
+        await CreateAgent()
+        
+        if (deviceManager == nil)
+        {
+            return
+        }
+        
+        videoDeviceInfoList = deviceManager?.cameras ?? []
+        videoDeviceInfoItemList = [VideoDeviceInfoItem]();
+        selectedVideoDeviceInfoIndex = videoDeviceInfoList.count > 0 ? 0 : -1;
+        
+        for i in 0 ..< videoDeviceInfoList.count
+        {
+            let videoDeviceInfoItem = VideoDeviceInfoItem(id: i, name: videoDeviceInfoList[i].name)
+            videoDeviceInfoItemList.append(videoDeviceInfoItem)
+        }
```

Comment:
Nit: Map is more nature to swift developers
```suggestion
        selectedVideoDeviceInfoIndex = videoDeviceInfoList.count > 0 ? 0 : -1;
        videoDeviceInforItemList = videoDeviceInfoList
             .enumerated()
             .map { (i, elt)  in VideoDeviceInfoItem(id: i, name: elt.name) }
```